### PR TITLE
New validation attributes

### PR DESCRIPTION
**Pull request by kpkozak**
- Content with same template
- XhtmlString block allowed types

### CODE REVIEW

File: EpiCommonUtils/Infrastructure/Validation/ContentWithSameTemplateAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                return ValidationResult.Success;
+            
+            var contentArea = (ContentArea) value;
+
+            var isValid = contentArea.Items.Select(this.GetTemplate).Where(x => x != null).Select(x => x.TemplateType)
+                .Distinct().Count() <= 1;
+
+            return isValid
+                ? ValidationResult.Success
+                : new ValidationResult("Content on the list must be of the same type");
```
Comment (tomaszmadeyski) [inline EpiCommonUtils/Infrastructure/Validation/ContentWithSameTemplateAttribute.cs:28]: Maybe adding a list of different templates used to error message would make finding source or error easier?

File: EpiCommonUtils/Infrastructure/Validation/XHtmlStringAllowedBlockTypes.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+using EPiServer.Core;
+using EPiServer.Core.Html.StringParsing;
+using EPiServer.ServiceLocation;
+
+namespace Forte.EpiCommonUtils.Infrastructure.Validation
+{
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
+    public class XHtmlStringAllowedBlockTypes : ValidationAttribute
+    {
+        private readonly IContentLoader contentLoader = ServiceLocator.Current.GetInstance<IContentLoader>();
```
Comment (tomaszmadeyski) [inline EpiCommonUtils/Infrastructure/Validation/XHtmlStringAllowedBlockTypes.cs:14]: you use `private readonly IContentLoader contentLoader = ServiceLocator.Current.GetInstance<IContentLoader>();` here and `private Injected<IContentLoader> _contentLoader;` in other attribute. Maybe we can be consistent here?

Comment (kpkozak) [reply EpiCommonUtils/Infrastructure/Validation/XHtmlStringAllowedBlockTypes.cs:14]: What would be preferred one?

Comment (tomaszmadeyski) [reply EpiCommonUtils/Infrastructure/Validation/XHtmlStringAllowedBlockTypes.cs:14]: I would vote for Injected but it's not a strong prefference. Consistency is primary goal

**Pull request merged**